manticore/platforms/evm.py

```diff
@@ -1648,6 +1649,7 @@ def _close_transaction(self, result, data=None, rollback=False):
             self._logs = logs
             self.send_funds(tx.address, tx.caller, tx.value)
 
+        self.increase_nonce(tx.caller)
```

Comment:
Do nonces get incremented in rollbacked transactions? 
if not we need an `else:` there

Reply:
In the real world, I think the transaction still gets mined even if a TXERROR occurs, right? If so, then the nonce should increment. I will double-check.

Reply:
Failed Internal tx? Like internal tx with no enough funds ? Would those increment nonce?

Reply:
Oh, a _contract's_ nonce is only ever incremented if it _creates_ another contract. Internal transactions should not increment the nonce.

Reply:
Unless the internal tx create a contract I presume. 
What about account creation via CALL/SELFDESTRUCT to non existing accounts?
Ughh nonces are problematic.